src/util.c

```diff
@@ -487,55 +489,88 @@ systemd_escape(char* string)
 gboolean
 netplan_delete_connection(const char* id, const char* rootdir)
 {
-    g_autofree gchar* del = NULL;
+    g_autofree gchar* yaml_path = NULL;
     g_autoptr(GError) error = NULL;
     NetplanNetDefinition* nd = NULL;
+    FILE* patch = NULL;
     gboolean ret = TRUE;
 
-    NetplanState* np_state = netplan_state_new();
-    NetplanParser* npp = netplan_parser_new();
+    NetplanParser* input_parser = netplan_parser_new();
+    NetplanState* input_state = netplan_state_new();
+    NetplanParser* output_parser = NULL;
+    NetplanState* output_state = NULL;
 
     /* parse all YAML files */
-    if (   !netplan_parser_load_yaml_hierarchy(npp, rootdir, &error)
-        || !netplan_state_import_parser_results(np_state, npp, &error)) {
+    if (   !netplan_parser_load_yaml_hierarchy(input_parser, rootdir, &error)
+        || !netplan_state_import_parser_results(input_state, input_parser, &error)) {
+        g_warning("netplan_delete_connection: Cannot parse input: %s\n", error->message);
+        ret = FALSE;
+        goto cleanup;
+    }
+
+    /* find specified netdef in input state */
+    nd = netplan_state_get_netdef(input_state, id);
+    if (!nd) {
+        g_fprintf(stderr, "netplan_delete_connection: Cannot delete %s, does not exist.", id);
```

Comment:
You seem to have mix of calls to g_fprintf() with patterns [having / having no] trailing "\n".
I think at least for this occurrence, it is missing the "\n" (unless `g_fprintf` adds it automatically: I'm not familiar with glib).

I can't speak for the other occurrences where the pattern ends in "%s". You might expect that the parameter (i.e. error message) contains the the final "\n".

Also, you might want to change `"Cannot delete %s, does not exist/"` by `"Cannot delete %s: does not exist."` but this is beyond nitpick :)

Reply:
ACK. I was experimenting a bit with `g_warning()`/`g_error()`/g_fprintf()`, but `g_error()` actually terminates the application, which is not directly what we want here. So this was a left-over. Fixed now.